### ISSUE TITLE
Regarding the time difference between robot data and residential area

### ISSUE THREAD

**Issue by Sadcatcat** [freqtrade/freqtrade#10924]
## Describe your environment

  * Operating system: windows 11
  * Python Version:Python 3.12.7
  * CCXT version:ccxt==4.4.24
  * Freqtrade Version:freqtrade 2024.10
  
## Your question
You've worked hard, brother. You've helped me so much, and my gratitude is beyond words
I am in China and I am using a cloud server in Hong Kong because my OKX account's trading time is calculated based on domestic time and the Eastern Eight Districts. I have verified that my server time matches my account and my place of residence. Why is the data I see in FreqUI and Telegram delayed or not real-time? Although it's just a simulation trade,
Or is it because OKX's data does not allow real-time collection？

**server information：**

> root@ser575683352015:~/docker/ft_userdata# timedatectl status
>                       Local time: Wed 2024-11-13 04:07:47 CST
>                   Universal time: Tue 2024-11-12 20:07:47 UTC
>                         RTC time: Tue 2024-11-12 20:07:48
>                        Time zone: Asia/Shanghai (CST, +0800)
>        System clock synchronized: yes
> systemd-timesyncd.service active: yes
>                  RTC in local TZ: no
> root@ser575683352015:~/docker/ft_userdata# date
> Wed Nov 13 04:07:51 CST 2024
> root@ser575683352015:~/docker/ft_userdata# docker exec -it freqtrade /bin/bash
> ftuser@c71890feabd6:/freqtrade$ date
> Wed Nov 13 04:08:04 CST 2024

**Docker-compose.yml file**
Added environment: TZ: Asia/Shanghai parameter
```
---
services:
  freqtrade:
    image: freqtradeorg/freqtrade:stable
    environment:
      TZ: Asia/Shanghai
    # image: freqtradeorg/freqtrade:develop
    # Use plotting image
    # image: freqtradeorg/freqtrade:develop_plot
    # # Enable GPU Image and GPU Resources (only relevant for freqAI)
    # # Make sure to uncomment the whole deploy section
    # deploy:
    #   resources:
    #     reservations:
    #       devices:
    #         - driver: nvidia
    #           count: 1
    #           capabilities: [gpu]
    # Build step - only needed when additional dependencies are needed
    # build:
    #   context: .
    #   dockerfile: "./docker/Dockerfile.custom"
    restart: unless-stopped
    container_name: freqtrade
    volumes:
      - "./user_data:/freqtrade/user_data"
    # Expose api on port 8080 (localhost only)
    # Please read the https://www.freqtrade.io/en/stable/rest-api/ documentation
    # for more information.
    ports:
      - "0.0.0.0:8080:8080"
    # Default command used when running `docker compose up`
    command: >
      trade
      --logfile /freqtrade/user_data/logs/freqtrade.log
      --db-url sqlite:////freqtrade/user_data/tradesv3.sqlite
      --config /freqtrade/user_data/config.json
      --strategy MeanReversionLongStrategy

```

Why are these errors every time I refresh my FreqUI
**log：**
```
freqtrade  | 2024-11-13 04:02:33,258 - uvicorn.error - INFO - Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
freqtrade  | 2024-11-13 04:02:33,312 - freqtrade.resolvers.iresolver - INFO - Using resolved pairlist StaticPairList from '/freqtrade/freqtrade/plugins/pairlist/StaticPairList.py'...
freqtrade  | 2024-11-13 04:02:33,324 - freqtrade.plugins.pairlist.IPairList - WARNING - Pair MATIC/USDT:USDT is not compatible with exchange OKX. Removing it from whitelist..
freqtrade  | 2024-11-13 04:02:33,330 - freqtrade.plugins.pairlistmanager - INFO - Whitelist with 8 pairs: ['DOGE/USDT:USDT', 'SOL/USDT:USDT', 'ETH/USDT:USDT', 'BTC/USDT:USDT', 'LTC/USDT:USDT', 'XRP/USDT:USDT', 'BNB/USDT:USDT', 'DOT/USDT:USDT']
freqtrade  | 2024-11-13 04:02:33,332 - freqtrade.strategy.hyper - INFO - No params for buy found, using default values.
freqtrade  | 2024-11-13 04:02:33,332 - freqtrade.strategy.hyper - INFO - No params for sell found, using default values.
freqtrade  | 2024-11-13 04:02:33,333 - freqtrade.strategy.hyper - INFO - No params for protection found, using default values.
freqtrade  | 2024-11-13 04:02:33,333 - freqtrade.plugins.protectionmanager - INFO - No protection Handlers defined.
freqtrade  | 2024-11-13 04:02:33,333 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': status, 'status': 'running'}
freqtrade  | 2024-11-13 04:02:33,337 - freqtrade.worker - INFO - Changing state to: RUNNING
freqtrade  | 2024-11-13 04:02:33,351 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': warning, 'status': 'Dry run is enabled. All trades are simulated.'}
freqtrade  | 2024-11-13 04:02:33,351 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': startup, 'status': "*Exchange:* `okx`\n*Stake per trade:* `unlimited USDT`\n*Minimum ROI:* `{'0': 0.04, '30': 0.06, '120': 0.1}`\n*Trailing Stoploss:* `-0.1`\n*Position adjustment:* `Off`\n*Timeframe:* `5m`\n*Strategy:* `MeanReversionLongStrategy`"}
freqtrade  | 2024-11-13 04:02:33,352 - freqtrade.rpc.telegram - INFO - Notification 'startup' not sent.
freqtrade  | 2024-11-13 04:02:33,399 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': startup, 'status': "Searching for USDT pairs to buy and sell based on [{'StaticPairList': 'StaticPairList'}]"}
freqtrade  | 2024-11-13 04:02:33,399 - freqtrade.rpc.telegram - INFO - Notification 'startup' not sent.
freqtrade  | 2024-11-13 04:02:33,682 - telegram.ext.Application - INFO - Application started
freqtrade  | 2024-11-13 04:02:36,497 - uvicorn.error - INFO - ('87.254.23.2', 39397) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:02:36,497 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(04363232, ('87.254.23.2', 39397))
freqtrade  | 2024-11-13 04:02:36,498 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:02:36,978 - uvicorn.error - INFO - ('87.254.23.2', 46939) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:02:36,979 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(8417eb7c, ('87.254.23.2', 46939))
freqtrade  | 2024-11-13 04:02:36,979 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:02:38,409 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:02:54,573 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:02:54,574 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(04363232, ('87.254.23.2', 39397))
freqtrade  | 2024-11-13 04:02:54,584 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:02:54,584 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(8417eb7c, ('87.254.23.2', 46939))
freqtrade  | 2024-11-13 04:02:56,520 - uvicorn.error - INFO - ('87.254.23.2', 51883) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:02:56,521 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(79fdd51e, ('87.254.23.2', 51883))
freqtrade  | 2024-11-13 04:02:56,522 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:02:57,033 - uvicorn.error - INFO - ('87.254.23.2', 58439) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:02:57,033 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(e46374bc, ('87.254.23.2', 58439))
freqtrade  | 2024-11-13 04:02:57,034 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:03:38,414 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:03:56,355 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:03:56,356 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(79fdd51e, ('87.254.23.2', 51883))
freqtrade  | 2024-11-13 04:03:56,875 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:03:56,876 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(e46374bc, ('87.254.23.2', 58439))
freqtrade  | 2024-11-13 04:04:07,615 - uvicorn.error - INFO - ('87.254.23.2', 60077) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:04:07,616 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(16b8640c, ('87.254.23.2', 60077))
freqtrade  | 2024-11-13 04:04:07,617 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:04:07,630 - uvicorn.error - INFO - ('87.254.23.2', 35585) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:04:07,630 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(c29a60d8, ('87.254.23.2', 35585))
freqtrade  | 2024-11-13 04:04:07,631 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:04:38,418 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:05:37,450 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:05:37,451 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(16b8640c, ('87.254.23.2', 60077))
freqtrade  | 2024-11-13 04:05:37,466 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:05:37,466 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(c29a60d8, ('87.254.23.2', 35585))
freqtrade  | 2024-11-13 04:05:41,002 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:05:48,008 - uvicorn.error - INFO - ('87.254.23.2', 37557) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:05:48,008 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(45808a71, ('87.254.23.2', 37557))
freqtrade  | 2024-11-13 04:05:48,009 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:05:48,041 - uvicorn.error - INFO - ('87.254.23.2', 42295) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:05:48,041 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(9368db98, ('87.254.23.2', 42295))
freqtrade  | 2024-11-13 04:05:48,042 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:06:41,006 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:06:47,852 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:06:47,854 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(45808a71, ('87.254.23.2', 37557))
freqtrade  | 2024-11-13 04:06:47,875 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:06:47,876 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(9368db98, ('87.254.23.2', 42295))
freqtrade  | 2024-11-13 04:06:58,618 - uvicorn.error - INFO - ('87.254.23.2', 37617) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:06:58,619 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(dea9048b, ('87.254.23.2', 37617))
freqtrade  | 2024-11-13 04:06:58,620 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:06:58,637 - uvicorn.error - INFO - ('87.254.23.2', 32857) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:06:58,637 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(5089c0d9, ('87.254.23.2', 32857))
freqtrade  | 2024-11-13 04:06:58,638 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:07:41,011 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:07:58,460 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:07:58,462 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(dea9048b, ('87.254.23.2', 37617))
freqtrade  | 2024-11-13 04:07:58,482 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:07:58,483 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(5089c0d9, ('87.254.23.2', 32857))
freqtrade  | 2024-11-13 04:08:09,649 - uvicorn.error - INFO - ('87.254.23.2', 35325) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:08:09,650 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(0daac54c, ('87.254.23.2', 35325))
freqtrade  | 2024-11-13 04:08:09,651 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:08:09,657 - uvicorn.error - INFO - ('87.254.23.2', 55945) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" [accepted]
freqtrade  | 2024-11-13 04:08:09,657 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(9dc8cafb, ('87.254.23.2', 55945))
freqtrade  | 2024-11-13 04:08:09,658 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:08:41,016 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:09:09,398 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:09:09,400 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(0daac54c, ('87.254.23.2', 35325))
freqtrade  | 2024-11-13 04:09:09,408 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:09:09,409 - freqtrade.rpc.api_server.ws.channel - INFO - Disconnected from channel - WebSocketChannel(9dc8cafb, ('87.254.23.2', 55945))
freqtrade  | 2024-11-13 04:09:20,552 - uvicorn.error - INFO - ('87.254.23.2', 59161) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" 403
freqtrade  | 2024-11-13 04:09:20,554 - uvicorn.error - INFO - connection rejected (403 Forbidden)
freqtrade  | 2024-11-13 04:09:20,555 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:09:20,599 - uvicorn.error - INFO - ('87.254.23.2', 47725) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDIxMTgsImlhdCI6MTczMTQ0MTIxOCwidHlwZSI6ImFjY2VzcyJ9.OwQBPwIRQ-E_Uf6KPcpYEO4qcwkPaa9KC4DpMR66Jho" 403
freqtrade  | 2024-11-13 04:09:20,600 - uvicorn.error - INFO - connection rejected (403 Forbidden)
freqtrade  | 2024-11-13 04:09:20,601 - uvicorn.error - INFO - connection closed
freqtrade  | 2024-11-13 04:09:41,020 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:10:46,002 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:11:46,008 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:12:46,012 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:13:46,017 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
freqtrade  | 2024-11-13 04:14:36,345 - uvicorn.error - INFO - ('87.254.23.2', 35705) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDMwNjMsImlhdCI6MTczMTQ0MjE2MywidHlwZSI6ImFjY2VzcyJ9.MpsZYBRBVkqYU-aaJhy0Ntp2evkwPUeiAWeEK74fwKY" [accepted]
freqtrade  | 2024-11-13 04:14:36,347 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(100358a2, ('87.254.23.2', 35705))
freqtrade  | 2024-11-13 04:14:36,348 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:14:36,727 - uvicorn.error - INFO - ('87.254.23.2', 34689) - "WebSocket /api/v1/message/ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZGVudGl0eSI6eyJ1IjoiZnJlcXRyYWRlciJ9LCJleHAiOjE3MzE0NDMwNjMsImlhdCI6MTczMTQ0MjE2MywidHlwZSI6ImFjY2VzcyJ9.MpsZYBRBVkqYU-aaJhy0Ntp2evkwPUeiAWeEK74fwKY" [accepted]
freqtrade  | 2024-11-13 04:14:36,728 - freqtrade.rpc.api_server.ws.channel - INFO - Connected to channel - WebSocketChannel(edf4ee06, ('87.254.23.2', 34689))
freqtrade  | 2024-11-13 04:14:36,729 - uvicorn.error - INFO - connection open
freqtrade  | 2024-11-13 04:14:46,023 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2024.10', state='RUNNING'
```

**time difference**
For example, for me now, it's 4 o'clock on the 13th, but the data inside the robot is still around 8 o'clock on the 12th
![Image](https://github.com/user-attachments/assets/e647d9e9-f1d5-4300-b48c-6abde3be84a0)
![Image](https://github.com/user-attachments/assets/884e48a7-c766-4405-9ea8-df47b641eed2)

**Comment by xzmeng** [comment]
Currently, freqtrade uses UTC as the timezone, so 4:00 AM on the 13th in China corresponds to 8:00 PM on the 12th in UTC.

**Comment by Sadcatcat** [comment]
> Currently, freqtrade uses UTC as the timezone, so 4:00 AM on the 13th in China corresponds to 8:00 PM on the 12th in UTC.

Thank you, my friend. May I ask if this will impact my trading? Or, when the Freqtrade bot executes real trades and retrieves OKX API information, will it detect any abnormalities? During actual trading, does it mean that the bot's orders are placed as events that occurred in the past in my timezone?

**Comment by xzmeng** [comment]
Using UTC in Freqtrade generally shouldn’t impact your trading results, even though it may feel like events are occurring “in the past” from your timezone’s perspective. Here’s how it works:
	1.	Order Execution Timing: When the Freqtrade bot places trades, it uses UTC timestamps internally. This keeps all processes consistent across different timezones and ensures that trades align with real-time market data as provided by exchanges like OKX.
	2.	Data from the OKX API: The data retrieved from the OKX API is also typically timestamped in UTC or another standardized format. This means the bot’s data retrieval and trade execution are naturally in sync with the exchange, so no irregularities should occur due to timezone differences.
	3.	Viewing Data in Local Time: When you check the bot’s trading logs or other data in your local time (China Standard Time), it may appear as though trades happened “earlier” than the current local time. However, this is only a presentation issue, not one that impacts the timing or accuracy of trades.

In summary, while it can feel disorienting, the UTC timezone setting is a standard approach for avoiding timezone complications in trading, and it shouldn’t negatively affect your trades or cause any abnormal behavior.

_Answered by chatgpt_

**Comment by xzmeng** [comment]
TL;DR: It’s only a presentation issue; it won’t affect anything else.

**Comment by xmatthias** [comment]
Whenever the UI is refreshed - the websocket will reconnect, causing a couple of messages (disconnect, reconnect).

As for the timezone - internally, the bot will only use UTC timezone - which is aligned to how exchange api's work.
You can modify the log timestamp (passing TZ to the container,...) - but that won't affect much other than logs and maybe a few messages.

As for the UI:
In the FreqUI settings, you can select the timezone, which will either be UTC, or your local timezone.

![Image](https://github.com/user-attachments/assets/9e1be66c-45f0-4222-b952-a908636c2391)
